tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_DuDt.cpp

```diff
@@ -8,13 +8,274 @@
 #include <random>
 
 #include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Identity.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Mesh.hpp"
 #include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/PointwiseFunctions/GeneralRelativity/TestHelpers.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
 #include "Utilities/Gsl.hpp"
 
 // IWYU pragma: no_forward_declare Tensor
 
 namespace {
+// Since repeatedly generating new numbers to compare to SpEC when the function
+// signature of the GH time derivative changes is difficult, we use a reference
+// implementation that we test against the Spectral Einstein Code (SpEC) to test
+// against our RHS changes.
```

Comment:
I'm sure this is fine, but could you please help me understand what's going on here? Specifically, where did this reference implementation come from? Is it just copied from spec? If not, where did it come from?

I get that you first test the reference implementation against spec, and then test the spectre implementation against the reference implementation, but I don't get where the reference implementation came from or how it differs from just coding up the RHS again, basically copying code from src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

Reply:
Ah, it was just copied from the old `src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp` which was tested against SpEC. I kept the test against SpEC because I think it's useful to have as a comparison case and will make things like adding extra constraint damping parameters easier in the future. Does that make sense and sound reasonable?